experimental/iterators/include/iterators/Dialect/Iterators/IR/IteratorsOps.td

```diff
@@ -104,6 +104,58 @@ class LookupFuncType<string opName, string symbolName>
             "    " # symbolName # ".dyn_cast<FlatSymbolRefAttr>())"
             "       .getFunctionType()">;
 
+/// A predicate, that is, a function with a signature of the form (T) -> i1.
+class Iterators_Predicate
```

Comment:
I'd rename these 2 PredicateFunctionType + PredicateFunctionSymbol to be more idiomatic.